clash-lib/src/Clash/Netlist/Types.hs

```diff
@@ -532,6 +532,13 @@ data Declaration
       !BlackBox                -- ^ Template tokens
       BlackBoxContext          -- ^ Context in which tokens should be rendered
 
+  -- | @component@ declaration (VHDL).
+  --
+  -- See [this reference](https://www.ics.uci.edu/~jmoorkan/vhdlref/compdec.html).
```

Comment:
A better pointer would be to section 4.5 of IEEE 1076-1993, i.e. the version of the VHDL standard we target in Clash